nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -190,4 +203,118 @@ public static Optional<Conflict> extractSingleConflict(
     Conflict conflict = conflicts.get(0);
     return Optional.of(conflict);
   }
+
+  public static ViewMetadata loadViewMetadata(
+      ViewMetadata metadata, String metadataLocation, Reference reference) {
+    Map<String, String> newProperties = Maps.newHashMap(metadata.properties());
+    newProperties.put(NessieTableOperations.NESSIE_COMMIT_ID_PROPERTY, reference.getHash());
+
+    return ViewMetadata.buildFrom(
+            ViewMetadata.buildFrom(metadata).setProperties(newProperties).build())
+        .setMetadataLocation(metadataLocation)
+        .build();
+  }
+
+  static void handleExceptionsForCommits(Exception exception, String refName, Content.Type type) {
```

Comment:
should this maybe also return `Optional<..>`?

Reply:
we always throw an exception from this method as it is called only for 3 types of exception which are filtered. 
So, I think it is not required here.

Reply:
it is more about how this is handling in the exception block where this method is called. Having an explicit `throw` where this method is called is more readable IMO